src/Utils/SchemaPrinter.php

```diff
@@ -487,4 +491,46 @@ public static function printIntrospectionSchema(Schema $schema, array $options =
             $options
         );
     }
+
+    public static function printSchemaDirectives(Type $type) : string
+    {
+        if ($type->astNode === null) {
+            return '';
+        }
+
+        if ($type->astNode instanceof ObjectTypeDefinitionNode) {
+            $directives = $type->astNode->directives;
+        } elseif ($type->astNode instanceof InterfaceTypeDefinitionNode) {
+            $directives = $type->astNode->directives;
+        } else {
+            return '';
+        }
+
+        if ($directives instanceof NodeList) {
+            $count      = $directives->count();
+            $directives = iterator_to_array($directives->getIterator());
+        } else {
+            $count = count($directives);
+        }
+
+        return $count > 0 ? (' ' . implode(
```

Comment:
I'd invert the condition so `else` is not hidden at the bottom: `' ' . $count === 0 ? '' : implode(`